identity/app/views/support/fragment/ConsentChannel.scala

```diff
@@ -33,5 +23,18 @@ object ConsentChannel {
     channels.filter(_._2).map(_._1)
   }
 
+  /** Field is a channel and user has provided details for this channel */
+  def isUsersChannel(consentField: Field, user: User): Boolean = {
+      consentField("id").value.exists { id =>
+        List("phone", "sms", "post").contains(id) && channelsProvidedBy(user).exists(_.channelName == id)
```

Comment:
`List("phone", "sms", "post")` can be a value in the object to avoid unnecessary allocation. You can also construct it using the ConsentChannelBehaviour to avoid duplication of the names.

Reply:
I agree with @rahilb here